library/core/src/ptr/const_ptr.rs

```diff
@@ -409,6 +409,20 @@ impl<T: ?Sized> *const T {
     #[rustc_const_stable(feature = "const_ptr_offset", since = "1.61.0")]
     #[inline(always)]
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
+    #[requires(
+        // Precondition 1: the computed offset `count * size_of::<T>()` does not overflow `isize`
+        count.checked_mul(core::mem::size_of::<T>() as isize).is_some() &&
+        // Precondition 2: adding the computed offset to `self` does not cause overflow
+        (self as isize).checked_add((count * core::mem::size_of::<T>() as isize)).is_some() &&
```

Comment:
I believe you could introduce a local variable to hold the result of `count.checked_mul(core::mem::size_of::<T>() as isize)`. This would enable avoiding multiplying twice (which is a costly operation in Kani).